resource/resource_test.go

```diff
@@ -189,6 +191,31 @@ func TestOut(t *testing.T) {
 			inDir, teardown := setup(t, defDir, sshRemote(cfg.Owner, cfg.Repo), cfg.SHA, cfg.SHA)
 			defer teardown(t)
 
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusCreated)
+				fmt.Fprintln(w, "Anything goes...")
+
+				if tc.want.body != nil {
+					buf, _ := io.ReadAll(r.Body)
+					var bm map[string]string
+					if err := json.Unmarshal(buf, &bm); err != nil {
+						t.Fatalf("parsing JSON body: %v", err)
+					}
+					for k, v := range tc.want.body {
```

Comment:
if the test case passes us a non-nil map, we decode the JSON object into the `bm` map and then validate that each k,v pair in `tc.want.body` (in this case: `map[string]string{"context": "cocco/" + defEnv.Get("BUILD_JOB_NAME")})` is also present in the JSON.